objects/src/accounts/component/template/storage/placeholder.rs

```diff
@@ -28,6 +28,26 @@ pub struct StoragePlaceholder {
     key: String,
 }
 
+/// An identifier for the expected type for a storage placeholder.
+/// These indicate which variant of [StorageValue] should be provided when instantiating a
+/// component.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum PlaceholderType {
+    Felt,
+    Map,
+    Word,
+}
```

Comment:
This may may be completely wrong as I didn't think through the structure deep enough, but would it make sense to make `StoragePlaceholder` itself typed? For example, something like this:
```rust
pub enum StoragePlaceholder {
    Felt(String),
    Map(String),
    Word(String),
}
```

Reply:
I thought of that initially but it seemed like it introduced a bit more complexity because now we have to additionally validate that storage entries are well-formed (ie, no internal field has a placeholder with an incorrect type). This obviously wouldn't happen if we are deserializing from toml because we can infer the correct type at the correct moment, but it does complicate serialization and deserialization a bit more as well because we couldn't necessarily rely on derived implementations or untagged enums such as with this definition of `MapRepresentation`:
```rust
pub enum MapRepresentation {
    List(Vec<MapEntry>),
    Template(StoragePlaceholder),
}
```
Also it felt a little redundant in general.

Reply:
The one disadvantage of this approach is that there is no way to tell if an `InitStorageData` will match with the component template's requirements

Reply:
By "this approach" do you mean the current approach or the approach I proposed?

Reply:
Oh, I meant the current one, sorry for the confusion